src/domain/orders/new/new-order.tsx

```diff
@@ -1,311 +1,130 @@
-    ])
-  }
-
-  const handlePriceChange = (price, index) => {
-    const value = Math.round(price * 100)
-
-    const updated = [...items]
-    updated[index] = {
-      ...items[index],
-      unit_price: value,
-    }
-
-    setItems(updated)
-  }
-
-  const handleAddQuantity = (quantity, index) => {
-    if (quantity < 0) {
-      return
-    }
-
-    const updated = [...items]
-    updated[index] = {
-      ...items[index],
-      quantity,
-    }
-
-    setItems(updated)
-  }
-
-  const handleRemoveItem = (index) => {
-    const updated = [...items]
-    updated.splice(index, 1)
-    setItems(updated)
-  }
-
-  const fetchShippingOptions = async (region) => {
-    try {
-      const { data } = await Medusa.shippingOptions.list({
-        region_id: region.id,
-        is_return: false,
-      })
-
-      setShippingOptions(data.shipping_options)
-      return data.shipping_options
-    } catch (error) {
-      throw Error("Could not fetch shipping options")
-    }
-  }
-
-  const getValidShippingOptions = () => {
-    const options = shippingOptions
-
-    const total = calculateTotal()
-
-    return options.reduce((acc, next) => {
-      if (next.requirements) {
-        const minSubtotal = next.requirements.find(
-          (req) => req.type === "min_subtotal"
-        )
-
-        if (minSubtotal) {
-          if (total <= minSubtotal.amount) {
-            return acc
-          }
-        }
-
-        const maxSubtotal = next.requirements.find(
-          (req) => req.type === "max_subtotal"
-        )
-        if (maxSubtotal) {
-          if (total >= maxSubtotal.amount) {
-            return acc
+    form: { handleSubmit, reset },
+    context: { region },
+  } = useNewOrderForm()
+
+  const onSubmit = handleSubmit((data) => {
+    mutate(
+      {
+        email: data.email,
+        items: data.items.map((i) => {
+          if (i.variant_id) {
+            return {
+              quantity: i.quantity,
+              variant_id: i.variant_id,
+            }
+          } else {
+            return {
+              quantity: i.quantity,
+              title: i.title,
+              unit_price: i.unit_price,
+            }
           }
-        }
-      }
-
-      acc.push(next)
-      return acc
-    }, [])
-  }
-
-  const submit = async () => {
-    const doItems = items.map((i) => {
-      const obj = {
-        variant_id: i.id || "",
-        quantity: i.quantity,
-        title: i.title,
+        }),
+        region_id: data.region.value,
+        shipping_methods: [
+          {
+            option_id: data.shipping_option.value,
+            price: data.custom_shipping_price
+              ? persistedPrice(
+                  region?.currency_code!,
+                  data.custom_shipping_price
+                )
+              : undefined,
+          },
+        ],
+        shipping_address: data.shipping_address_id
+          ? data.shipping_address_id
+          : !isNullishObject(data.shipping_address)
```

Comment:
This is a bit tricky because the API typing is wrong, it should accept either a id or a address. Also there is currently a bug i nthe core, that prevents the billing address to be added if it's not added as an id. I have an open PR to fix, that I need to get up to date an have merged.